WIP XuetangX/other CDN video player branding

clrux opened the pull request:
@frrrances  Mind reviewing the FED?
@andy-armstrong Mind reviewing the Jasmine tests here?
@cptvitamin Mind reviewing a11y?

--

CC FYI @polesye @tymofij @olmar

polesye commented on lms/templates/video.html:
```diff
@@ -101,45 +119,59 @@ <h3 class="hidden">${_('No playable video sources found.')}</h3>
+
+      % if branding_info:
+      <li id="branding" class="branding">
+        <!--
```
Please use `##` instead `<!--`.

polesye commented on lms/templates/video.html:
```diff
@@ -52,45 +52,63 @@
+                <li class="size-control"><a href="#" class="add-fullscreen" title="${_('Fill browser')}" role="button" aria-disabled="false"><i class="icon icon-fullscreen" aria-hidden="true"></i><span class="sr">${_('Fill browser')}</span></a></li>
+                <li class="quality-control"><a href="#" class="quality-toggle-control is-hidden" title="${_('HD off')}" role="button" aria-disabled="false">${_('HD off')}</a></li>
+              </ul>
+              <!-- Captioning/Subtitles -->
```
Please use `##` instead `<!--`.

polesye commented on lms/templates/video.html:
```diff
@@ -52,45 +52,63 @@
               </div>
-          </section>
-          <a class="nav-skip sr" id="before-transcript_${id}" href="#after-transcript_${id}">${_('Skip to end of transcript.')}</a>
+              <!-- Size and quality -->
```
Please use `##` instead `<!--`.

polesye commented on lms/templates/video.html:
```diff
@@ -52,45 +52,63 @@
+              <!-- Playback volume -->
+              <div class="volume">
+                <a href="#" role="button" aria-disabled="false" title="${_('Volume')}" aria-label="${_('Click on this button to mute or unmute this video or press UP or DOWN buttons to increase or decrease volume level.')}"><i class="icon icon-volume-up" aria-hidden="true"></i></a>
+                <!-- ### <a href="#" role="button" aria-disabled="false" title="${_('Volume')}" aria-label="${_('Click on this button to mute or unmute this video or press UP or DOWN buttons to increase or decrease volume level.')}"><i class="icon icon-volume-off" aria-hidden="true"></i></a> ### When audio is muted -->
```
Please use `##` instead `<!--`.

polesye commented on lms/templates/video.html:
```diff
@@ -52,45 +52,63 @@
+              <div class="speeds menu-container">
+                <a class="speed-button" href="#" title="${_('Speeds')}" role="button" aria-disabled="false">
+                  <i class="icon icon-caret-right" aria-hidden="true"></i>
+                  <!-- ### <i class="icon icon-caret-up" aria-hidden="true"></i>  ### On hover, when menu is visible-->
```
Please use [`##`](http://docs.makotemplates.org/en/latest/syntax.html#comments) instead `<!--`.

polesye commented on lms/templates/video.html:
```diff
@@ -52,45 +52,63 @@
+            </ul>
+            <!-- Playback controls -->
+            <div class="secondary-controls">
+              <!-- Playback speed -->
```
Please use [`##`](http://docs.makotemplates.org/en/latest/syntax.html#comments) instead `<!--`.

polesye commented on lms/templates/video.html:
```diff
@@ -52,45 +52,63 @@
+              <!-- ### <li class="video-control"><a class="video_control" href="#" title="${_('Pause')}" role="button" aria-disabled="false"><i class="icon icon-pause" aria-hidden="true"></i></a></li> ### When video is paused -->
+              <li class="video-control"><div class="vidtime">0:00 / 0:00</div></li>
+            </ul>
+            <!-- Playback controls -->
```
Please use [`##`](http://docs.makotemplates.org/en/latest/syntax.html#comments) instead `<!--`.

polesye commented on lms/templates/video.html:
```diff
@@ -52,45 +52,63 @@
+            <!-- Playback actions (play, pause, time) -->
+            <ul class="video-controls-options">
+              <li class="video-control"><a class="video_control" href="#" title="${_('Play')}" role="button" aria-disabled="false"><i class="icon icon-play" aria-hidden="true"></i></a></li>
+              <!-- ### <li class="video-control"><a class="video_control" href="#" title="${_('Pause')}" role="button" aria-disabled="false"><i class="icon icon-pause" aria-hidden="true"></i></a></li> ### When video is paused -->
```
Can it be removed?

clrux replied:
Yes, we can remove this line.

polesye commented on common/lib/xmodule/xmodule/js/src/video/09_video_caption.js:
```diff
@@ -789,7 +789,11 @@ function (Sjson, AsyncProcess) {
 
             hideSubtitlesEl
-                .attr('title', text)
+                .attr('title', gettext(text))
```
Why do we need `gettext` here? See lines 782 and 788.

polesye commented on common/lib/xmodule/xmodule/js/src/video/09_video_caption.js:
```diff
@@ -789,7 +789,11 @@ function (Sjson, AsyncProcess) {
-                .attr('title', text)
+                .attr('title', gettext(text))
+                .removeClass('show_transcript')
+                .removeClass('hide_transcript')
```
Why do we need to remove these class names? Does it really has these class names?

polesye replied:
```javascript
.removeClass('show_transcript hide_transcript')
```

polesye commented on common/lib/xmodule/xmodule/js/src/video/09_video_caption.js:
```diff
@@ -789,7 +789,11 @@ function (Sjson, AsyncProcess) {
+                .removeClass('hide_transcript')
+                .addClass(type)
+                .find('span')
                 .text(gettext(text));
```
Do not need to use `gettext` here.

polesye commented on common/lib/xmodule/xmodule/js/src/video/04_video_control.js:
```diff
@@ -238,7 +240,9 @@ function () {
-            .attr('title', gettext('Play'));
+            .attr('title', gettext('Play')).find('i')
+                .removeClass('icon-pause').
+                addClass('icon-play');
```
I think the following alignment is easier to read: 
```javascript
.attr('title', gettext('Play'))
.find('i')
  .removeClass('icon-pause')
  .addClass('icon-play');
```
What do you think?

clrux replied:
Agreed, that was a mistake on my part. Thanks for catching it.

polesye commented on common/lib/xmodule/xmodule/js/spec/video/video_caption_spec.js:
```diff
@@ -1091,6 +1093,8 @@
                     expect($('a.hide-subtitles'))
                         .toHaveAttr('title', 'Turn off captions');
+                        .toHaveClass('show_transcript');
+                        .toContain('i.icon.icon-quote-left');
```
Does it work?

polesye replied:
chain pattern

polesye commented on common/lib/xmodule/xmodule/js/spec/video/video_caption_spec.js:
```diff
@@ -1065,6 +1065,8 @@
                     expect($('a.hide-subtitles'))
                         .toHaveAttr('title', 'Turn on captions');
+                        .toHaveClass('hide_transcript');
+                        .toContain('i.icon.icon-quote-left');
```
Does it work?

polesye commented on common/lib/xmodule/xmodule/js/src/video/04_video_control.js:
```diff
@@ -226,7 +226,9 @@ function () {
-            .attr('title', gettext('Pause'));
+            .attr('title', gettext('Pause')).find('i')
+                .removeClass('icon-play').
+                addClass('icon-pause');
```
Fix alignment.

polesye commented on common/lib/xmodule/xmodule/js/src/video/07_video_volume_control.js:
```diff
@@ -193,10 +193,14 @@ function() {
+            this.el[action]('is-muted').find('a i')
+                .removeClass('icon-volume-up')
+                .removeClass('icon-volume-off')
+                .removeClass('icon-volume-undefined')
```
```javascript
.removeClass('icon-volume-up icon-volume-off icon-volume-undefined')
```
Fix alignment.

polesye commented on common/lib/xmodule/xmodule/css/video/video-player.scss:
```diff
@@ -0,0 +1,500 @@
+    -moz-font-smoothing: antialiased;
+    -ms-font-smoothing: antialiased;
+    -o-font-smoothing: antialiased;
+    font-smoothing: antialiased;
```
It'd be great to have a mixin for `font-smoothing`.

polesye commented on common/lib/xmodule/xmodule/css/video/video-player.scss:
```diff
@@ -0,0 +1,500 @@
+    -webkit-transition: -webkit-transform 0.7s ease-in-out;
+    -moz-transition: -moz-transform 0.7s ease-in-out;
+    -ms-transition: -ms-transform 0.7s ease-in-out;
+    transition: transform 0.7s ease-in-out;
```
```scss
@include transition(transform 0.7s ease-in-out);
```

polesye commented on common/lib/xmodule/xmodule/css/video/video-player.scss:
```diff
@@ -0,0 +1,500 @@
+        -webkit-transition: -webkit-transform 0.7s ease-in-out;
+        -moz-transition: -moz-transform 0.7s ease-in-out;
+        -ms-transition: -ms-transform 0.7s ease-in-out;
+        transition: transform 0.7s ease-in-out;
```
Please use mixin.

polesye commented on common/lib/xmodule/xmodule/css/video/video-player.scss:
```diff
@@ -0,0 +1,500 @@
+
+    .video-wrapper {
+        width: flex-grid(9,9);
+
```
Remove empty line.

polesye commented on common/lib/xmodule/xmodule/js/src/video/07_video_volume_control.js:
```diff
@@ -193,10 +193,14 @@ function() {
+            this.el[action]('is-muted').find('button i')
+                .removeClass('icon-volume-up')
+                .removeClass('icon-volume-off')
+                .removeClass('icon-volume-undefined')
```
Why do we need to remove all these class names?

polesye commented on common/lib/xmodule/xmodule/js/src/video/07_video_volume_control.js:
```diff
@@ -193,10 +193,14 @@ function() {
          * @param {Boolean} isMuted Flag to use muted or unmuted view.
          */
-        updateMuteButtonView: function(isMuted) {
+        updateMuteButtonView: function(isMuted, volumeIcon) {
```
Do we really need second argument? Can we rewrite it so?

```javascript
updateMuteButtonView: function(isMuted) {
  var volumeIcon = isMuted ? 'icon-volume-off' : 'icon-volume-up',
        action = isMuted ? 'addClass' : 'removeClass';
        this.el[action]('is-muted')
            .find('button i')
                .removeClass('icon-volume-up icon-volume-off')
                .addClass(volumeIcon);
}
```

polesye commented on common/lib/xmodule/xmodule/js/src/video/09_video_caption.js:
```diff
@@ -779,18 +779,22 @@ function (Sjson, AsyncProcess) {
-                .text(gettext(text));
+                .attr('title', gettext(text))
+                .removeClass('show_transcript')
+                .removeClass('hide_transcript')
```
`show_transcript` -> `show-transcript`
`hide_transcript` -> `hide-transcript`

polesye commented:
@clrux I see a few UI bugs on my laptop. 
* speed menu
![screen shot 2015-01-09 at 15 51 25](https://cloud.githubusercontent.com/assets/1247491/5680777/74c8c1d6-981a-11e4-98c4-2e890acdae1b.png)

* fullscreen button - just click.
![screen shot 2015-01-09 at 16 01 27](https://cloud.githubusercontent.com/assets/1247491/5680776/74c83130-981a-11e4-98d0-e024b7c89ead.png)
![screen shot 2015-01-09 at 16 01 33](https://cloud.githubusercontent.com/assets/1247491/5680778/74ca4344-981a-11e4-8f5c-cab711282e97.png)

clrux commented:
@polesye Will you pull down the latest version and check those icons again?

clrux commented:
@polesye I think all that's left is the Jasmine tests. When you looked, was everything alright?

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -42,20 +42,21 @@ div.video {
   .wrapper-downloads {
     margin: 0;
     padding: 0;
+    list-style: none;
```
Looks like you could use the %ui-no-list extend here.

clrux replied:
This doesn't work so just using the overrides here. Cleanup forthcoming in pattern library/cleanup.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -42,20 +42,21 @@ div.video {
 
       > a {
-        @include transition(all $tmg-f2 ease-in-out 0s);
+        @include transition(all 0.25s ease-in-out 0s);
```
why did you remove the timing variable?

clrux replied:
I don't think this was me, or it was me but unintentional.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -42,20 +42,21 @@ div.video {
     .video-download-button{
       display: inline-block;
       vertical-align: top;
-      margin: ($baseline*0.75) ($baseline/2) 0 0;
```
Our current coding practices ask for 0 in front of the multipliers less than 1.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -64,24 +65,51 @@ div.video {
+      vertical-align: top;
+
+      .host-tag {
+        // The @extend %cont-text-hide wasn't working for some reason...
```
Can you figure out why and note it here?

clrux replied:
Apparently this extend doesn't exist, even though it does. Advised to just use overrides in this case. Will cleanup later after pattern library and cleanup starts.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -42,20 +42,21 @@ div.video {
         border-radius: 3px;
         background-color: $very-light-text;
-        padding: ($baseline*0.75);
+        padding: ($baseline*.75);
```
Our current coding practices ask for 0 in front of the multipliers less than 1.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -64,24 +65,51 @@ div.video {
+    .branding {
+      display: inline-block;
+      float: right;
+      margin: 15px 0 0 10px;
```
Why not use the baseline variables here?

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -113,25 +141,25 @@ div.video {
         border-width: 30px 0 30px 50px;
-        border-color: transparent transparent transparent $white;
-        position: relative;
+        border-color: transparent transparent transparent #ffffff;
```
Use the $white variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -113,25 +141,25 @@ div.video {
       top: 50%;
       left: 50%;
       padding: 30px;
+      background: rgba(0, 0, 0, 0.7);
```
Use the $shadow-d2 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -89,23 +117,23 @@ div.video {
       left: 50%;
+      z-index: 1;
       padding: 30px;
+      background: rgba(0, 0, 0, 0.7);
```
Use the $shadow-d2 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -158,59 +186,51 @@ div.video {
-        -webkit-font-smoothing: antialiased;
-        box-shadow: 1px 0 0 #555, inset 1px 0 0 #555;
-        color: $white;
+        color: #fff;
```
Use the $white variable

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -158,59 +186,51 @@ div.video {
-        box-shadow: 1px 0 0 #555, inset 1px 0 0 #555;
-        color: $white;
+        color: #fff;
+        font-weight: 800;
```
Use the  %t-strong extend.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -158,59 +186,51 @@ div.video {
         border-width: 0 1px;
         border-style: solid;
-        border-color: $black;
+        border-color: #000;
```
Use $black variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -158,59 +186,51 @@ div.video {
         &:hover, &:focus {
           background-color: #444;
-          color: $white;
+          color: #fff;
```
Use $white variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -158,59 +186,51 @@ div.video {
         &:active,
         &:focus {
-          color: $white;
+          color: #fff;
```
Use $white variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -158,59 +186,51 @@ div.video {
+        -webkit-font-smoothing: antialiased;
 
         &:hover, &:focus {
           background-color: #444;
```
Use $gray-d2 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
-        }
+      .video-playback-controls {
+        @include clearfix();
+        background: #181818;
```
Use $gray-d4 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
+      .video-playback-controls {
+        @include clearfix();
+        background: #181818;
+        color: #f2f2f2;
```
Use $gray-l5 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
+          margin: 0;
+          padding: 0;
+          list-style: none;
+          border-right: 1px solid #666;
```
Use $gray-d1 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
+            a {
+              @extend %video-button;
+              width: 14px;
+              padding: 0 lh(.75);
```
Can we do away with this lh and maybe just give it an accurate width? Not a blocker, but would be nice...

frrrances replied:
Eh, now that I've gotten a little further along, I see that they all have the lh padding. We can save them all for a future cleanup story.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
+
+              &:focus {
+                position: relative;
+                z-index: 10000;
```
Use %ui-depth3 variable

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
-            }
+            .vidtime {
+              padding: 0 lh(.75);
+              font-weight: bold;
```
Use %t-strong extend.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
-              background-position: 16px -50px;
-            }
+          .video-control:first-child {
+            border-right: 1px solid #666;
```
Use $gray-d1 variable

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
+          display: block;
           position: relative;
+          float: left;
+          border-right: 1px solid #666;
```
Use $gray-d1 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
+              float: left;
+              padding: 0 lh(.25) 0 lh(.5);
+              font-size: em(14);
+              font-weight: normal;
```
Use @include font-size(14); mixin

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
+              font-weight: normal;
+              letter-spacing: 1px;
+              text-transform: uppercase;
+              color: #999;
```
Use the $gray-l1 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
+              float: left;
+              margin-bottom: 0;
+              padding: 0 lh(.5) 0 0;
+              font-weight: bold;
```
Use %t-strong extend.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
+              margin-bottom: 0;
+              padding: 0 lh(.5) 0 0;
+              font-weight: bold;
+              color: #fff;
```
Use $white variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
+            width: 127px;
+            bottom: 46px;
+            opacity: 0;
+            background-color: #444;
```
Use $gray-d2 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
-              box-shadow:  0 1px 0 #555;
-              border-bottom: 1px solid $black;
-              color: $white;
+              color: #fff;
```
Use $white variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
                 text-overflow: ellipsis;
                 white-space: nowrap;
+                border: 0;
+                color: #fff;
```
Use $white variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
-                &:hover, &:focus {
+                &:hover,
+                &:focus {
                   background-color: #666;
```
I know you didn't touch these lines, but all the states should be the same colors, so please use $gray-d2 and $gray-l2 variables here (or something more accessible since those colors are pretty close).

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -393,131 +448,91 @@ div.video {
-            }
+          &:focus {
+            position: relative;
+            z-index: 10000;
```
Use %ui-depth3 extend.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -393,131 +448,91 @@ div.video {
           position: relative;
+          display: block;
+          float: left;
+          border-right: 1px solid #666;
```
Use $gray-d2 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -393,131 +448,91 @@ div.video {
             height: 125px;
-            margin-left: -1px;
+            opacity: 0;
+            background-color: #444;
```
Use $gray-d2 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -393,131 +448,91 @@ div.video {
               width: 5px;
               margin: 14px auto;
+              border: 0;
               background: #666;
```
Use $gray-d1 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -527,48 +542,91 @@ div.video {
+          margin: 0;
+          padding: 0;
+          list-style: none;
+          border-right: 1px solid #666;
```
Use $gray-d1 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -527,48 +542,91 @@ div.video {
+              border: none;
+
+              &.active {
+                background-color: #F44;
```
Whoa what are these colors? Not a blocker but we should standardize/variable-ize these (if they are used... turquoise and red?!).

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -527,48 +542,91 @@ div.video {
+          display: block;
+          float: left;
+          height: 46px;
+          border-right: 1px solid #666;
```
Use $gray-d1 variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -527,48 +542,91 @@ div.video {
+            }
+
+            &.active {
+              background-color: #F44;
```
This looks like the same crazy styling... flagged for later cleanup.

clrux replied:
The JS switches class names depending on the state. I'm working on a video player (complete) overhaul in another branch so I'll be sure and clean this up there and make it more stateful. For this branch, I think this will need to stay. Good catch though.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -629,105 +688,111 @@ div.video {
-    border: 0;
-    bottom: 0;
+    position: fixed;
+    z-index: 9999;
```
Use %ui-depth3 extend.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -629,105 +688,111 @@ div.video {
-    top: 0;
-    width: 100%;
     vertical-align: middle;
+    background: rgba(#000, .95);
```
Use $black variable.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -736,19 +801,19 @@ div.video {
-        @extend %ui-depth4;
+      .video-controls {
+        position: absolute;
+        z-index: 9999;
```
Use %ui-depth4 extend.

clrux replied:
I don't think they're needed any longer.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -257,131 +276,167 @@ div.video {
-          .menu {
+          .video-speeds {
             @include transition(none);
-            @extend %ui-depth1;
```
Why remove the depth extend?

clrux replied:
Don't think I was the one who removed it, but all but a few properties here were removed. Looks like it was cleaned up? Z-indexing doesn't appear to be an issue so I'll leave it off.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -393,131 +448,91 @@ div.video {
 
           .volume-slider-container {
             @include transition(none);
-            @extend %ui-depth1;
```
Why remove the depth extend?

clrux replied:
See above comment.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -629,105 +688,111 @@ div.video {
     }
 
-    ol.subtitles.html5 {
-        @extend %ui-depth0;
```
Why remove the depth extend?

clrux replied:
See above comment.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -629,105 +688,111 @@ div.video {
   }
 
   &.video-fullscreen {
-    @extend %ui-depth4;
```
Why remove the depth extend here?

clrux replied:
No longer necessary.

frrrances commented on common/lib/xmodule/xmodule/css/video/display.scss:
```diff
@@ -736,19 +801,19 @@ div.video {
       @include box-sizing(border-box);
       @include transition(none);
-      background: $black;
+      background: #000;
```
Why remove the color variable here?

frrrances commented on common/lib/xmodule/xmodule/js/fixtures/video_all.html:
```diff
@@ -26,38 +26,50 @@
+              <div class="slider" title="Video position"></div>
+              <div class="video-playback-controls">
+                <ul class="video-controls-options">
+                  <li class="video-control"><a class="video_control" href="#" title="Play" role="button" aria-disabled="false"><i class="icon fa fa-play" aria-hidden="true"></i></a></li>
```
nice classes!

frrrances commented on common/lib/xmodule/xmodule/js/fixtures/video_all.html:
```diff
@@ -26,38 +26,50 @@
-                    <a href="#" title="Volume" role="button" aria-disabled="false"></a>
-                    <div class="volume-slider-container">
-                      <div class="volume-slider"></div>
+                    <a href="#" role="button" aria-disabled="false" title="Volume" aria-label="${_('Click on this button to mute or unmute this video or press UP or DOWN buttons to increase or decrease volume level.')}"><i class="icon fa fa-volume-up" aria-hidden="true"></i></a>
```
Nice function note

frrrances commented:
Good work, @clrux! Overall, this is looking good. 

I have a couple of general comments about PRs: 

* Isolatable changes should be split into separate PRs (for example, in this case, the Xuetang branding and the accessibility/icon icon changes) to reduce risk and allow revert on either if there are any issues. I think in this case, you should probably cherry-pick the relevant commits into two separate PRs after the review is done.
* When making a big PR like this, please write a PR comment when you're ready for review that has JIRA ticket numbers, notes about what changes are included, instructions on how to set up an env to review (or create a sandbox that is prepped with content for review). That will help speed review and let Product join in as well.

For this PR, I made a bunch of code-comment-level feedback that I would like you to address before I give it thumbs: 

1. I made a bunch of comments around using variables - especially colors - so we can allow theming. 
2. I noticed that you removed a bunch of extends/mixins... if there is a general issue about using those here, you can add a note.
2. I am also seeing the speed bug that @polesye is seeing, though not the other ones... I think it would be good to set up a sandbox and make a video with all the options on to spot check that they are all working, and work well together.
3. This PR should be manually tested across all possible browsers since it is a core feature of every course.

Let me know when you've addressed my comments and I'll continue my review.

clrux commented:
@frrrances Please continue your review when you get a chance. Thanks!

clrux commented:
# Previewing
The (sandbox) [http://xuetangx.m.sandbox.edx.org] should have everything configured to demo the site/video player as if you're viewing from a location in China. Both the CMS and LMS use the same video player code and both have the 'CN' location forced meaning both platforms should show the branding for XuetangX.

All work (and then some) is derived from (JIRA UX-1232) [https://openedx.atlassian.net/browse/UX-1232].

## Updates included in this work
* Video player controls updated to use font icons (Font Awesome 4.2.0) replacing the background images of yesteryear.
* Refactored Sass accordingly
* Added better/proper accessibility to the new controls
* Appending branding information (as the last `<li>` in `.wrapper-downloads`) under the player

## Notables
* The branding info has been set in the JSON files per country code, so this branding can be applied to any country code that's added to the array (yay future-proofing)

# To-dos
* This PR contains additional work (icon font swapage and accompanying accessibility) which will be broken out into a separate branch of work for a separate review.
* I've not yet tested with developers (I wanted to give them a "final" state of things to work with), however since we're just appending data to the existing player based on user location, the added hit is minimal at best. Not to mention we're using icon fonts now rather than images which may offset any additional performance hit taken by the location query (which is necessary).
* I've got another branch in progress that revamps the player entirely and does some major markup modifications and cleanup, along with JS and Sass cleanup.

clrux commented:
Closing this branch in lieu of https://github.com/edx/edx-platform/pull/6575
